docs/creating-a-runtime-check.md

```diff
@@ -0,0 +1,181 @@
+# Creating a Runtime Check
+
+Runtime checks are used to perform tests by executing the plugins code.
+
+The Plugin Checker uses a number of interfaces to build out runtime checks.
+
+One interface is the `Runtime_Check` interface. This interface contains no methods, but is used to help identify a runtime check from static checks.
+
+In addtion, there is also the `Preparation` interface, which is used to define checks that have preparations. This interface defines a `prepare()` method which is used to run the logic required to prepare the environment before the check before is run.
+
+Both these interfaces are implemented in the `Abstract_Runtime_Check` class, which developers can use when building out their own runtime checks. This class defines both the `prepare()` and `run()` methods which are required to be implemented for every runtime check.
+
+Below is the basic scaffold when creating a custom runtime check.
+
+```php
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+
+class Custom_Check extends Abstract_Runtime_Check {
+  public function prepare() {
+    // Handle the Checks preparations and return a cleanup function.
+  }
+
+  public function run( Check_Results $results ) {
```

Comment:
There are some references to "results" throughout, which need to be corrected :)
```suggestion
  public function run( Check_Result $result ) {
```